lib/pages/timetable/view/events/event_view.dart

```diff
@@ -89,6 +90,72 @@ class _EventViewState extends State<EventView> {
               } else {
                 AppToast.show(S.of(context).errorPermissionDenied);
               }
+            }),
+        AppScaffoldAction(
+            icon: Icons.delete,
+            onPressed: () {
+              final user = Provider.of<AuthProvider>(context, listen: false)
+                  .currentUserFromCache;
+              if (!user.canEditPublicInfo) {
+                showDialog(
+                    context: context,
+                    builder: (BuildContext context) {
+                      return AlertDialog(
```

Comment:
When we reuse components throughout the app (like you've seen with the scaffold), we create a customizable widget in widgets/ and use that everywhere, for consistency. For dialogs, we have AppDialog defined, you should use that. (If it's missing an option you would need, you can always add a new parameter to it to make it more flexible)